hydroflow/src/util/mod.rs

```diff
@@ -110,10 +110,13 @@ where
     C: Default + Extend<S::Item>,
     S: Stream,
 {
-    let any = std::cell::Cell::new(true);
-    let mut unfused_iter = ready_iter(stream).inspect(|_| any.set(true));
+    use std::sync::atomic::Ordering;
+
+    let got_any_items = std::sync::atomic::AtomicBool::new(true);
+    let mut unfused_iter =
+        ready_iter(stream).inspect(|_| got_any_items.store(true, Ordering::Relaxed));
     let mut out = C::default();
-    while any.replace(false) {
+    while got_any_items.swap(false, Ordering::Relaxed) {
```

Comment:
This doesn't seem to change anything? Does it make the future `Send` or something?

Reply:
Yup needs to be `Sync` for the returned future to be `Send`